### DISCUSSION TITLE
Object Return Value for UseController with Custom Component

### DISCUSSION THREAD

**Discussion by darcho** [react-hook-form/react-hook-form#9031] [Q&A] [answered]
V7.35.0

I'm running into an issue with a custom component using an object as the value.

When I set the value to a Luxon DateTime object, it is setting the value correctly within the component, but the data returned in the submit is returning an object without the methods. Is this the proper way of defining the form? I used to use NestedValue and it would return the object properly, but it has now become deprecated.  Thanks!

```
interface Form {
  myDate: DateTime | null
}

const { handleSubmit, control, reset, formState } = useForm<Form>({
  defaultValues: {
    myDate: null,
  },    
});
  
const onSubmit: SubmitHandler<Form> = (data) => {
  console.log(data.myDate);  <--- returns object with no methods
}

```

**Comment by leapful** [comment, marked as answer]
@darcho 

When working with form values, the best practice is always using primitive type because the values are mostly serialized (transform into JSON string) to be sent remotely (HTTP, socket) therefore using custom object might lead to unexpected result. It has been warned in react-hook-form documentation:

>Avoid including custom object into the defaultValues. eg: moment, luxon as those will lead to unexpected result during internal object clone process.

Referrer: https://react-hook-form.com/api/useform

In your case, when setting a Luxon instance as field value, it will be transformed to string when value is set to DOM Input element and then alter the field value too.

Switching to primitive type like number for Date will solve the issue and just transform it to Luxon on the fly to perform comparison.

**Comment by Muffinmann** [comment]
Have you tried the 'setValueAs' option of the register function? https://react-hook-form.com/api/useform/register
which can wrap the DOM value into the one you want.
